### DISCUSSION TITLE
How to specify multiple CSS rules when extending theme (à la `fontSize`)?

### DISCUSSION THREAD

**Discussion by kokoshneta** [tailwindlabs/tailwindcss#9465] [Help] [answered]
Most Tailwind classes define only a single CSS rule, but a few define more than one. The `text-[size]` rules, for example, define both a font size and a line height.

In the [default configuration](https://github.com/tailwindlabs/tailwindcss/blob/master/stubs/defaultConfig.stub.js), we can see how this is done:

```
module.exports = {
  theme: {
    fontSize: {
      xs: ['0.75rem', { lineHeight: '1rem' }],
    }
  }
}
```

That is, the property itself is passed as an array where the first item is the simple value to be applied to the primary CSS property the class sets (here `font-size`) and subsequent items are objects that define both the property (`lineHeight`, in the same notation as used in the theme file itself) and the value.

I would like to replicate that when extending the theme. For example, I have a variety of colours named `primary` (depending on theme), all of which are relatively dark. When these are used as background colours, the text colour should always be the theme’s corresponding `contrast` colour; that is, the output should be the following (pretending the primary colour is just blue):

```
.bg-primary {
  background-color: blue;
  color: white;
}
```

Based on the syntax in the default configuration, I naïvely did this:

```
module.exports = {
  theme: {
    extend: {
      backgroundColor: {
        primary: ['blue', { textColor: 'white' }],
      },
    }
  }
}
```

But that doesn’t work. There are no errors, but the class `bg-primary` is not generated in the output file either. If I give a scalar value (just `'blue'`), it works fine, apart from the fact that the text colour doesn’t change, but all permutations of passing along an array fail.

Is there a way to do this (preferably without a plugin)? Or can theme extension not be used to define classes to set more than one CSS property at once?

**Comment by wongjn** [comment, marked as answer]
The `fontSize` additional property values are specific to the core `fontSize` plugin only and won't work with any other core property. There is no way to do this for other properties without using a plugin.

**Reply by kokoshneta** [reply]
Bugger! Plugin it is, then, I guess…
